src/stdlib_string_type.fypp

```diff
@@ -421,7 +421,11 @@ contains
         type(string_type), intent(in) :: string
         integer :: ich
 
-        ich = merge(ichar(string%raw(1:1)), 0, allocated(string%raw))
+        if (allocated(string%raw) .and. len_trim(string%raw) > 0) then
```

Comment:
`ichar(" ")` should be allowed, therefore it's better to use `len()` here. Also for safe access:
```suggestion
        if (allocated(string%raw) .and. len(string) > 0) then
```

Reply:
Thank you. Apply suggestions.